ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt

```diff
@@ -287,7 +251,21 @@ class GoogleMapsFragment : Hilt_GoogleMapsFragment(), Map {
     return checkNotNull(customCap)
   }
 
-  private fun onMapClick(latLng: LatLng) = handleAmbiguity(latLng)
+  private fun onMapClick(latLng: LatLng) =
+    lifecycleScope.launch(defaultDispatcher) {
+      val clickedPolygons = getPolygonFeaturesContaining(latLng)
+      if (clickedPolygons.isNotEmpty()) {
+        featureClicks.emit(clickedPolygons)
```

Comment:
Just curious: Why have we used `tryEmit` in some places and `emit` in others? Is there a benefit of using one over the other?

Reply:
`tryEmit()` doesn't suspend, and require a coroutine. It can theoretically overflow the buffer, but in practice in these cases there would be few items and would be unlikely. In cases where we were already in a coroutine, I used `emit()` since it's slightly safer.